Extract more meaningful modules from `model::cfd`

luckysori opened the pull request:
I set out to start tackling #1430, but I got carried away and started trying to remove things that didn't make sense any longer to me (such as the `Completed` enum) rather than just moving them around.

I intend to continue with #1430 in follow-up PRs. If reviewers already have some ideas for meaningful modules, that would be helpful, although I would action that feedback in the follow-up PRs.

luckysori committed:
Rename CfdEvent to EventKind

We already have a `cfd::Event` type in the `model` crate which
represents an instance of an event of a particular kind. This is why
renaming the `CfdEvent` to `EventKind` makes sense. It will also help
when removing the `model/cfd` submodule.

luckysori committed:
Rename cfd::Event to CfdEvent

So that when we remove the `model/cfd` submodule, the type still reads
the same.

luckysori committed:
Convert cfd module of model crate into an implementation detail

luckysori committed:
Extract hex_transaction module

luckysori committed:
Create contract_setup module

luckysori committed:
Create rollover module

luckysori committed:
Move RolloverProposal to maker_cfd.rs

The only place where it is used.

luckysori committed:
Remove unused SettlemendKind enum

luckysori committed:
Remove RolloverCompleted alias

We have not needed this type for a long time now, as we can simply
execute the correct command wherever we learn that the rollover has
succeeded, failed or been rejected.

For this we break down the `Cfd::roll_over` command into 3: one per
possible outcome.

luckysori committed:
Use command::Executor in collab_settlement_maker::Actor

Unfortunately the actor still needs a direct connection to the `db`,
because it has to call a method on the `Cfd` which is NOT a command.

luckysori committed:
Use command::Executor in collab_settlement_taker::Actor

Unfortunately the actor still needs a direct connection to the `db`,
because it has to call a method on the `Cfd` which is NOT a command.

luckysori committed:
Do not mutate when executing commands

luckysori committed:
Remove CollaborativeSettlementCompleted alias

We have not needed this type for a long time now, as we can simply
execute the correct command wherever we learn that the collaborative
settlement has succeeded, failed or been rejected.

For this we break down the `Cfd::settle_collaboratively` command into
3: one per possible outcome.

luckysori committed:
Remove unused Completed enum

thomaseizinger commented on model/src/hex_transaction.rs:
```diff
@@ -0,0 +1,50 @@
+use bdk::bitcoin::Transaction;
+use serde::Deserializer;
+use serde::Serializer;
+
```
We should usptream this.

luckysori replied:
Tracking this locally.

thomaseizinger commented on daemon/src/setup_maker.rs:
```diff
@@ -117,24 +115,45 @@ impl Actor {
             let _: Result<(), xtra::Disconnected> = match contract_future.await {
-                Ok(dlc) => this.send(SetupSucceeded { order_id, dlc }).await,
-                Err(error) => this.send(SetupFailed { order_id, error }).await,
+                Ok(dlc) => this.send(SetupSucceeded { dlc }).await,
+                Err(error) => this.send(SetupFailed { error }).await,
```
Tiny nit: This could have been a separate commit.

thomaseizinger reviewed (approved):
Nice work!

klochowicz reviewed (approved):
great refactor! thanks for giving much needed TLC to the `model` <3

rishflab reviewed (approved):
Thanks, will be much easier to navigate the codebase !

luckysori committed:
Remove OrderId from module-private message

The `setup_maker::Actor` doesn't need to be reminded of what `OrderId`
it corresponds to by messages that it sends to itself.

luckysori committed:
Remove SetupCompleted alias

We have not needed this type for a long time now, as we can simply
execute the correct command wherever we learn that the contract setup
has succeeded, failed or been rejected.

For this we break down the `Cfd::setup_contract` command into 3: one per
possible outcome.
